diff --git a/dist/cjs/load-matches.cjs b/dist/cjs/load-matches.cjs
index ab816e129d85a7a54c88371c5ba90c1dbec4cdd0..5e860f35e7fcd4d2f80d02856e1951999947b3ef 100644
--- a/dist/cjs/load-matches.cjs
+++ b/dist/cjs/load-matches.cjs
@@ -440,6 +440,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 				try {
 					await runLoader(inner, matchPromises, matchId, index, route);
 					const match = inner.router.getMatch(matchId);
+					if (!match) return;
 					match._nonReactive.loaderPromise?.resolve();
 					match._nonReactive.loadPromise?.resolve();
 					match._nonReactive.loaderPromise = void 0;
@@ -463,6 +464,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 		if (_tanstack_router_core_isServer.isServer ?? inner.router.isServer) return inner.router.getMatch(matchId);
 	} else {
 		const prevMatch = inner.router.getMatch(matchId);
+		if (!prevMatch) return inner.matches[index];
 		const activeIdAtIndex = inner.router.stores.matchesId.get()[index];
 		const previousRouteMatchId = (activeIdAtIndex && inner.router.stores.matchStores.get(activeIdAtIndex) || null)?.routeId === routeId ? activeIdAtIndex : inner.router.stores.matches.get().find((d) => d.routeId === routeId)?.id;
 		const preload = resolvePreload(inner, matchId);
@@ -470,6 +472,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 			if (prevMatch.status === "success" && !inner.sync && !prevMatch.preload && shouldReloadInBackground) return prevMatch;
 			await prevMatch._nonReactive.loaderPromise;
 			const match = inner.router.getMatch(matchId);
+			if (!match) return inner.matches[index];
 			const error = match._nonReactive.error || match.error;
 			if (error) handleRedirectAndNotFound(inner, match, error);
 			if (match.status === "pending") await handleLoader(preload, prevMatch, previousRouteMatchId, match, route);
@@ -485,6 +488,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 		}
 	}
 	const match = inner.router.getMatch(matchId);
+	if (!match) return inner.matches[index];
 	if (!loaderIsRunningAsync) {
 		match._nonReactive.loaderPromise?.resolve();
 		match._nonReactive.loadPromise?.resolve();
diff --git a/dist/esm/load-matches.js b/dist/esm/load-matches.js
index 3ad245aef7776899cdda3b6d6a84025cc759f9ad..1897ebcfaa842cb626fec3b6f04a5049ed1323a0 100644
--- a/dist/esm/load-matches.js
+++ b/dist/esm/load-matches.js
@@ -440,6 +440,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 				try {
 					await runLoader(inner, matchPromises, matchId, index, route);
 					const match = inner.router.getMatch(matchId);
+					if (!match) return;
 					match._nonReactive.loaderPromise?.resolve();
 					match._nonReactive.loadPromise?.resolve();
 					match._nonReactive.loaderPromise = void 0;
@@ -463,6 +464,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 		if (isServer ?? inner.router.isServer) return inner.router.getMatch(matchId);
 	} else {
 		const prevMatch = inner.router.getMatch(matchId);
+		if (!prevMatch) return inner.matches[index];
 		const activeIdAtIndex = inner.router.stores.matchesId.get()[index];
 		const previousRouteMatchId = (activeIdAtIndex && inner.router.stores.matchStores.get(activeIdAtIndex) || null)?.routeId === routeId ? activeIdAtIndex : inner.router.stores.matches.get().find((d) => d.routeId === routeId)?.id;
 		const preload = resolvePreload(inner, matchId);
@@ -470,6 +472,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 			if (prevMatch.status === "success" && !inner.sync && !prevMatch.preload && shouldReloadInBackground) return prevMatch;
 			await prevMatch._nonReactive.loaderPromise;
 			const match = inner.router.getMatch(matchId);
+			if (!match) return inner.matches[index];
 			const error = match._nonReactive.error || match.error;
 			if (error) handleRedirectAndNotFound(inner, match, error);
 			if (match.status === "pending") await handleLoader(preload, prevMatch, previousRouteMatchId, match, route);
@@ -485,6 +488,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 		}
 	}
 	const match = inner.router.getMatch(matchId);
+	if (!match) return inner.matches[index];
 	if (!loaderIsRunningAsync) {
 		match._nonReactive.loaderPromise?.resolve();
 		match._nonReactive.loadPromise?.resolve();
